### PR TITLE
Prime 354  create blog in oscar dashboard views

### PR DESCRIPTION
**Pull request by noraratp**

### CODE REVIEW

File: my_store/blog/dashboard/apps.py [hunk 1]
```diff
@@ -2,8 +2,7 @@
```
Comment (surasakp) [inline my_store/blog/dashboard/apps.py:2]: I think you should change to use _ to - in url-name such as blog_list_view => blog-list-view

File: my_store/blog/dashboard/forms.py [hunk 2]
```diff
@@ -12,3 +16,38 @@ def clean(self):
...
+
+class CategorySearchForm(forms.Form):
+    name = forms.CharField(required=False, label=("Name"))
+
+    class Meta:
+        fields = ['name']
+
+    def clean(self):
+        cleaned_data = super(CategorySearchForm, self).clean()
+        cleaned_data['name'] = cleaned_data['name']
```
Comment (surasakp) [inline my_store/blog/dashboard/forms.py]: Why do you like this?

File: my_store/blog/dashboard/tables.py [hunk 3]
```diff
@@ -7,6 +7,16 @@ class PostTable(tables.Table):
 
     author = tables.TemplateColumn('{{ record.author.email | default:"-" }}')
 
+    title = tables.TemplateColumn(
```
Comment (surasakp) [inline my_store/blog/dashboard/tables.py]: I think you should use LinkColumn. Please read this doc https://django-tables2.readthedocs.io/en/latest/pages/api-reference.html#django_tables2.columns.LinkColumn

File: my_store/blog/dashboard/views.py [hunk 4]
```diff
@@ -31,9 +37,127 @@ def apply_search(self, queryset):
...
+            instance=self.object)
+        return ctx
+
+    def process_all_forms(self, form):
+        if form.is_valid():
+            self.object = form.save(commit=False)
+        formset = self.category_formset_class(
+            self.request.POST, instance=self.object)
+        is_valid = form.is_valid() and formset.is_valid()
+        print(form.errors)
```
Comment (surasakp) [inline my_store/blog/dashboard/views.py]: Please remove this line

File: my_store/blog/dashboard/views.py [hunk 5]
```diff
@@ -31,9 +37,127 @@ def apply_search(self, queryset):
...
+
+class BlogDetailView(generic.UpdateView):
+    template_name = 'dashboard/blog/detail.html'
+    form_class = PostForm
+    context_object_name = 'post_detail'
+    category_formset_class = CategoryGroupFormSet
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        ctx['form'] = self.form_class(instance=self.object)
```
Comment (surasakp) [inline my_store/blog/dashboard/views.py]: Please remove this line. UpdateView had form context

File: my_store/blog/dashboard/views.py [hunk 6]
```diff
@@ -31,9 +37,127 @@ def apply_search(self, queryset):
...
+    category_formset_class = CategoryGroupFormSet
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        ctx['category_formset'] = self.category_formset_class(
+            instance=self.object)
+        return ctx
+
+    def process_all_forms(self, form):
+        if form.is_valid():
```
Comment (surasakp) [inline my_store/blog/dashboard/views.py]: Please remove it duplicate for check form valid in line 69

File: my_store/blog/templates/dashboard/blog/blog_row_title.html [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{% load django_tables2 %}
```
Comment (surasakp) [inline my_store/blog/templates/dashboard/blog/blog_row_title.html]: Remove this file we don't need to use

**Review by surasakp — COMMENTED** [review]
Can you test form and formset?

File: my_store/blog/tests/test_views.py [hunk 8]
```diff
@@ -83,3 +85,106 @@ def test_search_with_title_and_author_should_return_correctly_data(self):
...
+            'postcategory_set-2-category': [''],
+            'postcategory_set-2-post': [''],
+            'postcategory_set-2-id': [''],
+            'action': ['save']
+        }
+
+        response = self.client.post(
+            self.url, data=data)
+        self.assertEqual(response.status_code, 302)
+
```
Comment (gorsutlame) [inline my_store/blog/tests/test_views.py:129]: I think should test redirect with follow true

File: my_store/blog/tests/test_views.py [hunk 9]
```diff
@@ -83,3 +85,106 @@ def test_search_with_title_and_author_should_return_correctly_data(self):
...
+            'featured_image': self.util.create_image(),
+            'postcategory_set-TOTAL_FORMS': '2',
+            'postcategory_set-INITIAL_FORMS': '0',
+            'postcategory_set-MIN_NUM_FORMS': '1',
+            'postcategory_set-0-category': self.category_factory.id
+        }
+
+        response = self.client.post(self.url_post_detail_view, data=data)
+        self.assertEqual(response.status_code, 302)
+
```
Comment (gorsutlame) [inline my_store/blog/tests/test_views.py:180]: I think should test redirect with follow true

**Review by gorsutlame — APPROVED** [review]

**Pull request merged**